### PR TITLE
Header: Center logo above header if all 3 items are visible

### LINKED ISSUE #358: The site title on the blog page is not a link when a static homepage is set
**Describe the bug**
When a static page is set as the homepage, and a page is selected as the blog, the site title is not a link on the blog page

**To Reproduce**
1. Go to the customizer or the homepage setting and set pages to the homepage and blog.
2. Open the blog page, and see that the site title is not a link.

**Expected behavior**
The blog page is not the homepage so there needs to be a link.

### LINKED ISSUE #398: Site navigation should centre aligned to Site Branding section
**Describe the bug**
Site navigation(Primary Navigation) is not horizontally aligned to Site Branding Section.

**To Reproduce**
1. Go to 'Appearance - Menu`
2. Create Primary Navigation
3. Go to Customizer and add Logo
4. Refresh the page
5. See Primary Navigation and Site Branding section you will see the difference

**Expected behavior**
Primary Navigation should centre aligned with the Site Branding section

**Screenshots**
https://a.cl.ly/Qwuk0e4B

### PR DESCRIPTION
**Pull request by ryelle**
## Summary
If the logo, site title/tagline, and nav menu are all visible, the logo should be centered above the header. If the title/tagline is hidden, the logo should stay to the left.

This also incorporates changes from #362 & #406 — so it also fixes #358 & fixes #398

## Relevant technical choices:

This is using `row-gap` on `site-header` to add extra spacing between the site title/tagline & menu when the menu is long enough to wrap under the title. This allows for better vertical centering vs using a margin, which pushes the nav down. This is not supported in Safari (yet, appears to be on the way), or IE11. 

Vertical alignment with the current `margin-bottom` approach:
Screen Shot 2020-10-12 at 3 42 38 PM

See below for the `row-gap` alignment in this branch.

On Safari, if you have a long menu, you don't get a space anymore:

Screen Shot 2020-10-12 at 4 38 26 PM

So this is a trade-off, but IMO the improved vertical alignment outweighs the edge case of a long menu + older browser, especially since it's not unreadable.

## Test instructions

This PR can be tested by following these steps:
1. Toggle menus, site titles, and logos.
2. Check that it looks like you expect.

## Screenshots

All taken on Mac/Firefox.

1. Has a logo, menu, and title is visible

    screenshot-1

2. Has a logo, menu, and title is not visible

    screenshot-2

3. Has a logo, does not have a menu, and title is visible

    screenshot-3

4. Has a logo, does not have a menu, and title is not visible

    screenshot-4

5. No logo, has menu, and title is visible

    screenshot-5

6. No logo, has menu, and title is not visible

    screenshot-6

7. No logo, does not have a menu, and title is visible

    screenshot-7

8. No logo, does not have a menu, and title is not visible

    screenshot-8

9. Long menu

    screenshot-9

## Quality assurance
* [x] I have thought about any security implications this code might add.
* [x] I have checked that this code doesn't impact performance (greatly).
* [x] I have tested this code to the best of my abilities
* [x] I have checked that this code does not affect the accessibility negatively

### CODE REVIEW

**Comment by melchoyce** [comment]
Thank you for this very thorough PR!

> So this is a trade-off, but IMO the improved vertical alignment outweighs the edge case of a long menu + older browser, especially since it's not unreadable.

I agree 👍

**Comment by carolinan** [comment]
The screenshots look great but I thought you discarded this design weeks ago which is why it was not implemented. I am confused again.

File: assets/sass/06-components/header.scss [hunk 1]
```diff
@@ -4,6 +4,11 @@
 	display: flex;
 	align-items: flex-start;
 	flex-wrap: wrap;
+	row-gap: var(--global--spacing-vertical); // Add space in case the menu wraps below the site branding.
```
Comment (aristath) [inline assets/sass/06-components/header.scss:7]: This should be tested on Safari... Though `row-gap` has great browser-support on grid layouts, Safari doesn't support it for flex layouts.

**Review by aristath — CHANGES_REQUESTED** [review]
Code is sensible, makes sense and works. This looks OK to me :+1:

File: assets/sass/06-components/header.scss (hunk 1 above)
Comment (carolinan) [reply assets/sass/06-components/header.scss:7]: There is already a screenshot from Safari above :)

Comment (aristath) [reply assets/sass/06-components/header.scss:7]: Ah you're right, I missed that. I went straight to the code and missed the OP discussion :facepalm:  :+1:
